### PR TITLE
Replace DocComment tree with dedicated JSDocConstructor tree

### PR DESCRIPTION
**Pull request by gzm0**
This simplifies pretty much all usage sites.

### CODE REVIEW

**Comment by gzm0** [comment]
Draft because it depends on #4920.

**Review by sjrd — APPROVED** [review]
LGTM after #4920 gets merged and this is rebased.

**Comment by gzm0** [comment]
Rebased. Ready to be merged once CI passes.

**Pull request merged**
